BlackNectar/BlackNectar/SearchFilterViewController.swift

```diff
@@ -0,0 +1,343 @@
+        
+    }
+    
+    private func loadStores() {
+        
+        UserLocation.instance.requestLocation { coordinate in
+            
+            self.loadStoresInMapView(at: coordinate)
+            
+        }
+        
+    }
+    
+    private func prepareMapView() {
+        
+        mapView.delegate = self
+        mapView.showsUserLocation = true
+        
+        guard let region = UserLocation.instance.currentRegion else {
+            
+            LOG.error("Failed to the Users Current Region")
+            return
+        }
+        
+        self.mapView.setRegion(region, animated: true)
+        
+    }
+    
+}
+
+//MARK: Loading Stores into Map View
+extension SearchFilterViewController {
+    
+    func loadStoresInMapView(at coordinate: CLLocationCoordinate2D) {
+        
+        startSpinningIndicator()
+        
+        let distanceInMeters = DistanceCalculation.milesToMeters(miles: Double(distance))
+        
+        SearchStores.searchForStoresLocations(near: coordinate, with: distanceInMeters) { stores in
+            
+            self.stores = stores
+            
+            self.main.addOperation {
+                
+                self.populateStoreAnnotations()
+                self.stopSpinningIndicator()
+                
+            }
+            
+        }
+        
+        if self.stores.isEmpty {
+            
+            self.makeNoteThatNoStoresFound(additionalMessage: "User is in the Search Filter Map View")
+            
+        }
+        
+    }
+    
+    func populateStoreAnnotations() {
+        
+        var annotations: [MKAnnotation] = []
+        
+        for store in stores {
+            
+            let annotation = createAnnotation(forStore: store)
+            annotations.append(annotation)
+        }
+        
+        mapView.addAnnotations(annotations)
+        mapView.removeNonVisibleAnnotations()
+        
+    }
+    
+    func createAnnotation(forStore store: Store) -> CustomAnnotation {
+        
+        let storeName = store.storeName
+        let location = store.location
+        
+        let latitude = location.latitude
+        let longitude = location.longitude
+        
+        let annotation = CustomAnnotation(name: storeName, latitude: latitude, longitude: longitude)
+        
+        return annotation
+        
+    }
+    
+}
+
+//MARK: Map View Delegate Code
+extension SearchFilterViewController {
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        
+        if annotation is MKUserLocation {
+            
+            return nil
+            
+        }
+        
+        let annotation = annotation as? CustomAnnotation
+        let smallSquare = CGSize(width: 30, height: 30)
+        let button = UIButton(frame: CGRect(origin: CGPoint.zero, size: smallSquare))
+        let blackNectarPin = UIImage(named: "BlackNectarMapPin")
```

Comment:
You can probably create the UIImage once in the class, and then reuse it. It's never really gonna change.